### DISCUSSION TITLE
Res.json is not a function

### DISCUSSION THREAD

**Discussion by jalajgoel** [vercel/next.js#11735] [Help]
I am trying to use getStaticProps and in that I am fetching data from firebase, so converting it's response to json is throwing error.

**getStaticProps**

```js
export async function getStaticProps() {
  let res = await getListings();
  if (!!res) {
    // res = { data: res }
    console.log("res", res)
    const post = await res.json();
    return {
      props: { listing: null },
    };
  }
}
```

**getListings() Function**

```js
export const getListings = async () => {
  try {
    let query = firebase.collection("listings");
    query = query.orderBy("LastChangeTimestamp", "desc"); //Matrix_Unique_ID
return await query
      .limit(10)
      .get()
      .then(async (res) => {
      return res.docs.map(doc => {
            return doc.data()
        });
  } catch (error) {
    console.error("error getListings =>", error);
    return error;
  }
};
```

**Error is**

> res.json() is not a function.

>

**Comment by Timer** [comment]
You are calling a non-existent method in `getStaticProps`. The data is already available under the `.data` key, as seen via your comment.

**Comment by HarjaapSinghM** [comment]
@jalajgoel did Timer's response help? I'm trying to figure out this exact issue and don't fully understand the answer offered above
